tests/core/tools/style.js

```diff
@@ -128,6 +128,34 @@
 
 		'test style.parse.margin docs sample': function() {
 			objectAssert.areEqual( { top: '3px', right: '0', bottom: '2', left: '0' }, CKEDITOR.tools.style.parse.margin( '3px 0 2' ) );
+		},
+
+		'test style.parse.border docs sample': function() {
+			objectAssert.areEqual( { width: '3px', style: 'solid', color: '#ffeedd' }, CKEDITOR.tools.style.parse.border( '3px solid #ffeedd' ) );
+		},
+
+		'test style.parse.border only with width': function() {
+			objectAssert.areEqual( { width: '0%' }, CKEDITOR.tools.style.parse.border( '0%' ) );
+		},
+
+		'test style.parse.border only with zero width': function() {
+			objectAssert.areEqual( { width: '0' }, CKEDITOR.tools.style.parse.border( '0' ) );
+		},
+
+		'test style.parse.border with width and style': function() {
+			objectAssert.areEqual( { width: '0%', style: 'groove' }, CKEDITOR.tools.style.parse.border( '0% groove' ) );
+		},
+
+		'test style.parse.border with mixed color and style': function() {
+			objectAssert.areEqual( { color: '#ff0000', style: 'dotted' }, CKEDITOR.tools.style.parse.border( '#ff0000 dotted' ) );
+		},
+
+		'test style.parse.border with mixed color, style and width': function() {
+			objectAssert.areEqual( { width: '7.5em', color: '#ff0000', style: 'dotted' }, CKEDITOR.tools.style.parse.border( '#ff0000 dotted 7.5em' ) );
+		},
+
+		'test style.parse.border with style only': function() {
```

Comment:
Tests checking different color formats (`rgba`, `hsla`, color names) should be added (or you may modify existing tests so each contains different format).